src/main/java/com/cachinginapp/enterprise/CachingInController.java

```diff
@@ -79,10 +89,15 @@ public List<Cache> fetchAllCaches(){
      */
     @GetMapping("/cache/{id}/")
     public ResponseEntity fetchCacheById(@PathVariable("id") int id){
-        Cache foundCache = cacheService.fetchCacheById(id);
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
-        return new ResponseEntity(foundCache, headers, HttpStatus.OK);
+        try {
+            Cache foundCache = cacheService.fetchCacheById(id);
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
+            return new ResponseEntity(foundCache, headers, HttpStatus.OK);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "Unable to fetch cache with ID: " + id + ", message: " + e.getMessage());
+            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
```

Comment:
added headers, try catch, and logger to notify the user that the id that is being requested is unable to be fetched.